frontends/web/src/routes/account/walletconnect/utils.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright 2023 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Core } from '@walletconnect/core';
+import { IWeb3Wallet, Web3Wallet } from '@walletconnect/web3wallet';
+import { useCallback, useEffect, useState } from 'react';
+
+export let web3wallet: IWeb3Wallet;
+
+export const SUPPORTED_CHAINS = [
+  'eip155:1', // ETH Mainnet
+  'eip155:5', // ETH Goerli testnet
+  'eip155:10', // Optimism
+  'eip155:56', // BSC
+  'eip155:137', // Polygon
+  'eip155:250', // Fantom
+  'eip155:42161' // Arbitrum One
+];
+
+export const EIP155_SIGNING_METHODS = {
+  PERSONAL_SIGN: 'personal_sign',
+  ETH_SIGN: 'eth_sign',
+  ETH_SIGN_TRANSACTION: 'eth_signTransaction',
+  ETH_SIGN_TYPED_DATA: 'eth_signTypedData',
+  ETH_SIGN_TYPED_DATA_V3: 'eth_signTypedData_v3',
+  ETH_SIGN_TYPED_DATA_V4: 'eth_signTypedData_v4',
```

Comment:
This is a bit trickier than you'd think as with many poorly implemented ethereum things. The reality is that many dapps will request an 'eth_signTypedData' but in reality expect 'eth_signTypedData_v4' because that's basically the standard now and v4 is assumed, but not necessarily asked for specifically. 
We could probably get rid of  eth_signTypedData_v3 since I don't think anybody uses that specifically, though I would err on the side of caution, leave this as is, and assume all sign typed data requests are asking for v4 and return that

Reply:
cool. please add this comment as a comment in the code there, very helpful :smile: